contributors/devel/godep.md

```diff
@@ -1,127 +1,178 @@
-# make sure there is no client-go or other staging repo in $GOPATH before running the next command.
-./hack/update-staging-godeps.sh
-git checkout -- $(git status -s | grep "^ D" | awk '{print $2}' | grep ^Godeps)
+hack/run-in-gopath.sh hack/godep-save.sh
 ```
 
-_If `go get -u path/to/dependency` fails with compilation errors, instead try
-`go get -d -u path/to/dependency` to fetch the dependencies without compiling
-them. This is unusual, but has been observed._
+This will run through all of the primary targets for the Kubernetes project,
+calculate which deps are needed, and rebuild the database.  It will also
+regenerate other metadata files which the project needs, such as BUILD files and
+the LICENSE database.
 
 After all of this is done, `git status` should show you what files have been
-modified and added/removed.  Make sure to `git add` and `git rm` them.  It is
-commonly advised to make one `git commit` which includes just the dependency
-update and Godeps files, and another `git commit` that includes changes to
-Kubernetes code to use the new/updated dependency.  These commits can go into a
-single pull request.
+modified and added/removed.  Make sure to sanity-check them with `git diff`, and
+to `git add` and `git rm` them, as needed.  It is commonly advised to make one
+`git commit` which includes just the dependencies and Godeps files, and
+another `git commit` that includes changes to Kubernetes code to use (or stop
+using) the new/updated/removed dependency.  These commits can go into a single
+pull request.
 
-5) Before sending your PR, it's a good idea to sanity check that your
-Godeps.json file and the contents of `vendor/ `are ok by running `hack/verify-godeps.sh`
+## Sanity checking
 
-_If `hack/verify-godeps.sh` fails after a `godep update`, it is possible that a
-transitive dependency was added or removed but not updated by godeps. It then
-may be necessary to perform a `hack/godep-save.sh` to pick up the transitive
-dependency changes._
+Before sending your PR, it's a good idea to sanity check that your
+Godeps.json file and the contents of `vendor/ `are ok:
 
-It is sometimes expedient to manually fix the /Godeps/Godeps.json file to
-minimize the changes. However without great care this can lead to failures
-with `hack/verify-godeps.sh`. This must pass for every PR.
+```sh
+hack/run-in-gopath.sh hack/verify-godeps.sh
+```
 
-6) If you updated the Godeps, please also update `Godeps/LICENSES` by running
-`hack/update-godep-licenses.sh`.
+All this script will do is a restore, followed by a save, and then look for
+changes.  If you followed the above instructions, it should be clean.  If it is
+not, you get to figure out why.
 
+## Manual updates
+
+It is sometimes expedient to manually fix the `Godeps.json` file to
+minimize the changes. However without great care this can lead to failures
+with `hack/verify-godeps.sh`. This must pass for every PR.
 
```

Comment:
Probably we need a link or a mention that these scripts make symlinks for the code stored under the staging/ directory.

Reply:
ACK